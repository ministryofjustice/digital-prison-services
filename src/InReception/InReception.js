import React from 'react'
import PropTypes from 'prop-types'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import SortableColumn from '../tablesorting/SortableColumn'
import { LAST_NAME } from '../tablesorting/sortColumns'
import OffenderLink from '../OffenderLink'
import OffenderImage from '../OffenderImage'
import OffenderName from '../OffenderName'
import DateFormatter from '../DateFormatter'
import Flags from '../FullFlags/Flags'
import SortLov from '../tablesorting/SortLov'

const InReception = ({ sortOrder, setColumnSort, rows }) => (
  <>
    <GridRow>
      <GridCol setWidth="one-quarter">
        <SortLov sortColumns={[LAST_NAME]} sortColumn={LAST_NAME} sortOrder={sortOrder} setColumnSort={setColumnSort} />
      </GridCol>
    </GridRow>
    <table>
      <thead>
        <tr>
          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
          <th className="straight width10" />
          <th className="straight width15">
            <SortableColumn
              heading="Name"
              column={LAST_NAME}
              sortOrder={sortOrder}
              setColumnSort={setColumnSort}
              sortColumn={LAST_NAME}
            />
          </th>
          <th className="straight width10">Prison no.</th>
          <th className="straight width10">D.O.B.</th>
          <th className="straight width15"> Received from </th>
          <th className="straight width10">Incentive Level </th>
          <th className="straight width25">Flags</th>
        </tr>
      </thead>
      <tbody>
        {rows.map(row => (
          <tr className="row-gutters" key={row.offenderNo}>
            <td className="row-gutters">
              <OffenderLink offenderNo={row.offenderNo}>
                <OffenderImage offenderNo={row.offenderNo} />
              </OffenderLink>
            </td>
            <td className="row-gutters">
              <OffenderLink offenderNo={row.offenderNo}>
                <OffenderName lastName={row.lastName} firstName={row.firstName} />
              </OffenderLink>
            </td>
            <td className="row-gutters">{row.offenderNo}</td>
            <td className="row-gutters">
              <DateFormatter isoDate={row.dateOfBirth} />
            </td>
            <td> {row.fromAgencyDescription || row.fromCity} </td>
            <td> {row.iepLevel} </td>
            <td className="row-gutters">
              <Flags offenderNo={row.offenderNo} alerts={row.alerts} />
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </>
)

InReception.defaultProps = {
  sortOrder: '',
  setColumnSort: () => {},
  rows: [],
}

InReception.propTypes = {
  sortOrder: PropTypes.string,
  setColumnSort: PropTypes.func,
  rows: PropTypes.arrayOf(
    PropTypes.shape({
      offenderNo: PropTypes.string.isRequired,
      firstName: PropTypes.string.isRequired,
      fromAgencyDescription: PropTypes.string,
      dateOfBirth: PropTypes.string.isRequired,
      alerts: PropTypes.arrayOf(PropTypes.string),
      iepLevel: PropTypes.string,
      fromCity: PropTypes.string,
    })
  ),
}

export default InReception
