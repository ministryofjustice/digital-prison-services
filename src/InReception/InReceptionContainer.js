import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import axios from 'axios'

import { resetError, setLoaded } from '../redux/actions'
import Page from '../Components/Page'
import InReception from './InReception'
import routePaths from '../routePaths'
import SortableDataSource from '../tablesorting/SortableDataSource'

import { ASC } from '../tablesorting/sortOrder'
import { lastNameComparator } from '../tablesorting/comparatorComposition'

class InReceptionContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      offendersInReception: [],
    }
  }

  async componentDidMount() {
    const { agencyId, resetErrorDispatch, setLoadedDispatch, handleError } = this.props

    resetErrorDispatch()
    setLoadedDispatch(false)

    try {
      const response = await axios.get(`/api/movements/${agencyId}/in-reception`)

      this.setState({
        offendersInReception: response.data,
      })

      setLoadedDispatch(true)
    } catch (error) {
      handleError(error)
    }
  }

  componentDidUpdate(previousProps) {
    const { agencyId, history } = this.props

    if (agencyId !== previousProps.agencyId) {
      history.push(routePaths.establishmentRoll)
    }
  }

  render() {
    const { offendersInReception } = this.state

    return (
      <div>
        <Page title="In reception">
          <SortableDataSource sortOrder={ASC} rows={offendersInReception} comparator={lastNameComparator}>
            <InReception />
          </SortableDataSource>
        </Page>
      </div>
    )
  }
}

InReceptionContainer.propTypes = {
  agencyId: PropTypes.string.isRequired,
  history: PropTypes.shape({
    replace: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
  }).isRequired,
  resetErrorDispatch: PropTypes.func.isRequired,
  setLoadedDispatch: PropTypes.func.isRequired,
  handleError: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  agencyId: state.app.user.activeCaseLoadId,
})

const mapDispatchToProps = dispatch => ({
  resetErrorDispatch: () => dispatch(resetError()),
  setLoadedDispatch: status => dispatch(setLoaded(status)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(InReceptionContainer)
