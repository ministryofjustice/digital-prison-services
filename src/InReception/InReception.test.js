import React from 'react'
import { shallow } from 'enzyme'
import InReception from './InReception'

describe('In Reception', () => {
  it('should render the page correctly', () => {
    const inReception = [
      {
        offenderNo: 'offenderNo',
        firstName: 'firstName',
        lastName: 'lastName',
        timeOut: '12:00',
        fromAgencyDescription: 'Leeds',
        dateOfBirth: '1980-10-10',
        alerts: ['HA', 'XSA', 'XA', 'PEEP', 'XEL', 'XRF', 'XTACT'],
        iepLevel: 'Entry',
      },
    ]
    const wrapper = shallow(<InReception setColumnSort={jest.fn()} sortOrder="ASC" rows={inReception} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should display fromCity when fromAgencyDescription is not set ', () => {
    const inReception = [
      {
        offenderNo: 'offenderNo',
        firstName: 'firstName',
        lastName: 'lastName',
        timeOut: '12:00',
        fromCity: 'York',
        dateOfBirth: '1980-10-10',
        alerts: ['HA', 'XSA', 'XA', 'PEEP', 'XEL', 'XRF', 'XTACT'],
        iepLevel: 'Entry',
      },
    ]
    const wrapper = shallow(<InReception setColumnSort={jest.fn()} sortOrder="ASC" rows={inReception} />)

    expect(wrapper).toMatchSnapshot()
  })
})
