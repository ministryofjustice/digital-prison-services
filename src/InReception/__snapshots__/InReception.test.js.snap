// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In Reception should display fromCity when fromAgencyDescription is not set  1`] = `
<Fragment>
  <GridRow>
    <GridCol
      columnFull={false}
      columnOneHalf={false}
      columnOneQuarter={false}
      columnOneThird={false}
      columnThreeQuarters={false}
      columnTwoThirds={false}
      setWidth="one-quarter"
    >
      <SortLov
        setColumnSort={[MockFunction]}
        sortColumn="lastName"
        sortColumns={
          Array [
            "lastName",
          ]
        }
        sortOrder="ASC"
      />
    </GridCol>
  </GridRow>
  <table>
    <thead>
      <tr>
        <th
          className="straight width10"
        />
        <th
          className="straight width15"
        >
          <SortableColumn
            column="lastName"
            heading="Name"
            setColumnSort={[MockFunction]}
            sortColumn="lastName"
            sortOrder="ASC"
          />
        </th>
        <th
          className="straight width10"
        >
          Prison no.
        </th>
        <th
          className="straight width10"
        >
          D.O.B.
        </th>
        <th
          className="straight width15"
        >
           Received from 
        </th>
        <th
          className="straight width10"
        >
          Incentive Level 
        </th>
        <th
          className="straight width25"
        >
          Flags
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="row-gutters"
        key="offenderNo"
      >
        <td
          className="row-gutters"
        >
          <OffenderLink
            alerts={false}
            offenderNo="offenderNo"
          >
            <OffenderImage
              offenderNo="offenderNo"
            />
          </OffenderLink>
        </td>
        <td
          className="row-gutters"
        >
          <OffenderLink
            alerts={false}
            offenderNo="offenderNo"
          >
            <OffenderName
              firstName="firstName"
              lastName="lastName"
            />
          </OffenderLink>
        </td>
        <td
          className="row-gutters"
        >
          offenderNo
        </td>
        <td
          className="row-gutters"
        >
          <FormattedDate
            isoDate="1980-10-10"
          />
        </td>
        <td>
           
          York
           
        </td>
        <td>
           
          Entry
           
        </td>
        <td
          className="row-gutters"
        >
          <Flags
            alerts={
              Array [
                "HA",
                "XSA",
                "XA",
                "PEEP",
                "XEL",
                "XRF",
                "XTACT",
              ]
            }
            category=""
            offenderNo="offenderNo"
          />
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;

exports[`In Reception should render the page correctly 1`] = `
<Fragment>
  <GridRow>
    <GridCol
      columnFull={false}
      columnOneHalf={false}
      columnOneQuarter={false}
      columnOneThird={false}
      columnThreeQuarters={false}
      columnTwoThirds={false}
      setWidth="one-quarter"
    >
      <SortLov
        setColumnSort={[MockFunction]}
        sortColumn="lastName"
        sortColumns={
          Array [
            "lastName",
          ]
        }
        sortOrder="ASC"
      />
    </GridCol>
  </GridRow>
  <table>
    <thead>
      <tr>
        <th
          className="straight width10"
        />
        <th
          className="straight width15"
        >
          <SortableColumn
            column="lastName"
            heading="Name"
            setColumnSort={[MockFunction]}
            sortColumn="lastName"
            sortOrder="ASC"
          />
        </th>
        <th
          className="straight width10"
        >
          Prison no.
        </th>
        <th
          className="straight width10"
        >
          D.O.B.
        </th>
        <th
          className="straight width15"
        >
           Received from 
        </th>
        <th
          className="straight width10"
        >
          Incentive Level 
        </th>
        <th
          className="straight width25"
        >
          Flags
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="row-gutters"
        key="offenderNo"
      >
        <td
          className="row-gutters"
        >
          <OffenderLink
            alerts={false}
            offenderNo="offenderNo"
          >
            <OffenderImage
              offenderNo="offenderNo"
            />
          </OffenderLink>
        </td>
        <td
          className="row-gutters"
        >
          <OffenderLink
            alerts={false}
            offenderNo="offenderNo"
          >
            <OffenderName
              firstName="firstName"
              lastName="lastName"
            />
          </OffenderLink>
        </td>
        <td
          className="row-gutters"
        >
          offenderNo
        </td>
        <td
          className="row-gutters"
        >
          <FormattedDate
            isoDate="1980-10-10"
          />
        </td>
        <td>
           
          Leeds
           
        </td>
        <td>
           
          Entry
           
        </td>
        <td
          className="row-gutters"
        >
          <Flags
            alerts={
              Array [
                "HA",
                "XSA",
                "XA",
                "PEEP",
                "XEL",
                "XRF",
                "XTACT",
              ]
            }
            category=""
            offenderNo="offenderNo"
          />
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
